rfc/java11.md

```diff
@@ -0,0 +1,164 @@
+Java 8 JRE. So the first (more complicated) step is to get everyone migrated to
+a Java 11 JRE. Only after that is done can we switch to the Java 11 JDK.
+
+The Java 11 JRE has some fairly large changes, most notably the deprecation of
+the long-standing CMS garbage collector. Large organizations with highly-tuned
+JVMs have shown hesitancy to upgrade until the new runtime has undergone
+extensive testing. (Both Google and Netflix are in this category.)
+
+For official Spinnaker containers, we control the JRE used to run Spinnaker. For
+those releases, this proposal takes a relatively slow ramp to the Java 11 JRE
+and provides an escape hatch back to Java 8 for users who experience problems.
+
+There are also some unknown number of customers who aren't using the Spinnaker-
+provided containers. We will need to do some outreach to make sure these users
+convert to a Java 11 JRE too.
+
+## Timeline
+
+### 1.18 (early January)
+
+For our container builds, we'll first move just two (of approximately ten)
+services to the new JRE. Thee two services will be `igor` and `orca`. By
+default, anyone who uses Halyard to deploy these containers will get the Java 11
+JRE for these services. Users who experience problems or aren't ready to use
+Java 11 can use the `java8` variant by specifying `imageVariant: slim-java8` (or
+`ubuntu-java8`) in their Halyard config file.
+
+The release notes will prominently mention this fact. It will also be mentioned
+in-channel in the release announcement to `#spinnaker-releases` and in the
+release announcement email. We will strongly encourage people who opt for the
+`java8` variants to contact `sig-platform@spinnaker.io` or `#sig-platform` to
+discuss their concerns, which will allow us to adjust the timeline as needed.
+
+Separately, postings to `#dev` will encourage developers to start running their
+development environments under Java 11. Explanations of how to do so will be
+posted to the channel.
+
+If issues are found in Spinnaker that are caused by running under a Java 11
+runtime, we will switch the containers back to Java 8 and shift the timeline
+back one release. Since we're providing the `java8` variants, we will not adjust
+the timeline for any other reason.
+
+### 1.19 (early March)
+
+_All_ containers built by the build process will now use the Java 11 JRE. As
+before, the `java8` variants will offer a way to opt out of this change.
+
+Other than that, everything from the previous section applies to this release as
+well.
+
+### 1.20 (early May)
+
+For this release, we will do two things:
+
+1.  Compile with JDK 11 using the `-source 8 -target 8` flags. This produces
+    builds that are still compatible with Java 8, and means the build will still
+    fail if anyone tries to use Java 11 APIs or language features.
+2.  Remove the `java8` container variants
+
+If the compiler change causes issues, we can revert that change and push that
+out to the next release.
```

Comment:
How does this relate to the above statement of `we will not adjust the timeline for any other reason`? I assume it means that we won't adjust the _JRE_ timeline for any other reason and that the _JDK_ timeline is separate.

Reply:
I didn't really mean for that line to apply here. I pushed a commit that tries to clarify it more, let me know if it makes sense...